### PR TITLE
fix(graph/bfs): implement reasonable python queue

### PR DESCRIPTION
**Pull request by jhnlynn**
In Python, 'queue.Queue' is a 'multi-producer, multi-consumer queues. It is especially useful in threaded programming when information must be exchanged safely between multiple threads'.

If the 'Queue' is empty, and the programming is running, it’ll always waiting for `put`s, which may cause 'Time Limit Exceeds' in online coding platforms, such as Leetcode. 

Besides, using 'queue.Queue' is much slower than using `collections.deque` (which I changed it to) for the data structure `queue`.

P.S: Maybe code's not pythonic enough, because I tend to keep it, for beginners or people switched python to understand.

### CODE REVIEW

**Review by Menci — APPROVED** [review]
I agree to use `collections.deque`.

**Review by Enter-tainer — DISMISSED** [review]
nice catch!

File: docs/graph/bfs.md [hunk 1]
```diff
@@ -74,20 +74,20 @@ Python：
...
-    Q = Queue()
-    Q.put(u)
+    Q = deque() # common implementation of queue in Python
+    Q.append(u)
     vis[u] = True
     d[u] = 0
     p[u] = -1
-    while Q.qsize() != 0:
-        u = Q.get()
+    while len() != 0:
```
Comment (Backl1ght) [inline docs/graph/bfs.md:85]: len() -> len(Q)?

File: docs/graph/bfs.md [hunk 2]
```diff
@@ -74,20 +74,20 @@ Python：
...
+        u = Q.popleft()
         while i:
```
Comment (Backl1ght) [inline docs/graph/bfs.md:87]: 这个i是什么东西啊

File: docs/graph/bfs.md (hunk 1 above)
Comment (jhnlynn) [reply docs/graph/bfs.md:85]: Yep, it should be `len(Q)` in fact

File: docs/graph/bfs.md (hunk 2 above)
Comment (jhnlynn) [reply docs/graph/bfs.md:87]: 这是原始代码，我没有看到最近的 PR 对其进行了修改，所以我就不管它了.

Comment (Menci) [reply docs/graph/bfs.md:87]: i 应该先赋初值吧，否则会报未定义？

**Pull request closed without merging**